### PR TITLE
:sparkles: Add questionnaire view driven from example yaml

### LINKED ISSUE #1275: Questionnaire view
It will be possible for the Administrator to get a rendered view of the questionnaire once it listed in the questionnaires table. Clicking on the View option from the kebab menu of a row from the questionnaires table will navigate to a new view that renders the given questionnaire:

Image

As in the assessment view from the application inventory, questions are arranged in different sections accessible trough a set of vertical tabs. Each question in a section is represented as an expandable section, which once expanded will display the available answers for that question:

Image

Risk levels (Red, Yellow, Green) will be displayed as icons on each answer under the Risk column. Answers that imply autotagging will have an additional section below the choice text that will include the list of tags to be applied using labels to render each tag. The same applies for answers that would be automatically selected, rendering the list of tags that would trigger the selection of the answer. Conditional questions will have a Conditional label before the formulation text:

Upon expanding a conditional question, an additional section will be displayed above the list of answers, providing information about the condition. This could be either Include if present or Skip if present followed by a list of tags rendered using labels.

### PR DESCRIPTION
**Pull request by ibolton336**
Resolves #1275 

Screenshot 2023-08-15 at 1 10 00 PM

### CODE REVIEW

File: client/src/app/pages/assessment-management/questionnaire/components/answer-table.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  });
+  const {
+    currentPageItems,
+    numRenderedColumns,
+    propHelpers: { tableProps, getThProps, getTdProps },
+  } = tableControls;
+  const mapRiskToStatus = (risk: string): IconedStatusPreset => {
+    switch (risk) {
+      case "green":
+        return "LowRisk";
```
Comment (gitdallas) [inline client/src/app/pages/assessment-management/questionnaire/components/answer-table.tsx:43]: if you wanted to not add to the `IconedStatus` component, you could just return from here...

"green" = `<IconedStatus preset="ok" />` (as it already matches that preset)
"unknown" = `<IconedStatus preset="unknown" />`
"red" = `<IconedStatus icon={<TimesCircleIcon />} status="danger" />`
"yellow" = `<IconedStatus icon={<WarningTriangleIcon />} status="warning" />`

and then line 88 could be just `{ mapRiskToStatus(answer.risk) }` .. although maybe rename to `getIconByRisk`

but more of a matter of preference and if we expect these presets to be reused

File: client/src/app/pages/assessment-management/questionnaire/questionnaire-page.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                        {section.name}
+                      </TabTitleText>
+                    }
+                  >
+                    <TabContent
+                      id={section.name}
+                      className="tab-content-container"
+                      style={{ flex: 1 }}
+                    >
+                      <Table
```
Comment (gitdallas) [inline client/src/app/pages/assessment-management/questionnaire/questionnaire-page.tsx:145]: using the new table, nice.

would it make sense to pull this into its own component?

**Review by gitdallas — APPROVED** [review]
gave a couple thoughts but lgtm otherwise

File: client/src/app/pages/assessment-management/questionnaire/components/answer-table.tsx (hunk 1 above)
Comment (ibolton336) [reply client/src/app/pages/assessment-management/questionnaire/components/answer-table.tsx:43]: Done!

File: client/src/app/pages/assessment-management/questionnaire/questionnaire-page.tsx (hunk 2 above)
Comment (ibolton336) [reply client/src/app/pages/assessment-management/questionnaire/questionnaire-page.tsx:145]: Done - much cleaner page level component now.

File: client/src/app/components/IconedStatus.tsx [hunk 3]
```diff
@@ -5,6 +5,7 @@ import CheckCircleIcon from "@patternfly/react-icons/dist/esm/icons/check-circle
 import TimesCircleIcon from "@patternfly/react-icons/dist/esm/icons/times-circle-icon";
 import InProgressIcon from "@patternfly/react-icons/dist/esm/icons/in-progress-icon";
 import ExclamationCircleIcon from "@patternfly/react-icons/dist/esm/icons/exclamation-circle-icon";
+import WarningTriangleIcon from "@patternfly/react-icons/dist/esm/icons/warning-triangle-icon";
```
Comment (gitdallas) [inline client/src/app/components/IconedStatus.tsx:8]: remnant

File: client/src/app/pages/assessment-management/questionnaire/components/answer-table.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    currentPageItems,
+    numRenderedColumns,
+    propHelpers: { tableProps, getThProps, getTdProps },
+  } = tableControls;
+
+  const getIconByRisk = (risk: string): React.ReactElement => {
+    switch (risk) {
+      case "green":
+        return <IconedStatus preset="Ok" />;
+      case "unknown":
```
Comment (gitdallas) [inline client/src/app/pages/assessment-management/questionnaire/components/answer-table.tsx:46]: nit really but could probably just let this fall into default

Comment (ibolton336) [reply client/src/app/pages/assessment-management/questionnaire/components/answer-table.tsx]: Done.

**Pull request merged**